GoogleDataTransport/GDTCORLibrary/GDTCORPlatform.m

```diff
@@ -118,6 +121,51 @@ GDTCORNetworkMobileSubtype GDTCORNetworkMobileSubTypeMessage() {
 #endif
 }
 
+void GDTCOREncodeArchive(id<NSSecureCoding> obj, NSString *archivePath, NSError *_Nullable *error) {
+#if (defined(__IPHONE_11_0) && __IPHONE_OS_VERSION_MAX_ALLOWED >= 110000) || \
+    (defined(__MAC_10_13) && MAC_OS_X_VERSION_MAX_ALLOWED >= 101300) ||      \
+    (defined(__TVOS_11_0) && __TV_OS_VERSION_MAX_ALLOWED >= 110000)
+  if (@available(macOS 10.13, iOS 11.0, tvOS 11.0, *)) {
```

Comment:
Do we still need this if(@available(*)) check for version on macOS , iOS and tvOS? because we already have checked above. All other changes LGTM

Reply:
I think yes, we still do in order to handle cases like an app is compiled to be run on iOS 10 - 13 but is deployed on iOS 10 device. `if(@available(*))` is a runtime check vs the macros - compile time check.

Reply:
Got new knowledge! Thanks.^_^